src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs

```diff
@@ -43,29 +44,48 @@ public sealed class BasicAuthenticator : Authenticator
         /// Gets the username that this object holds
         /// </summary>
         [NotNull]
+        public string Username { get; }
+
+        /// <summary>
+        /// [DEPRECATED] Gets the password that this object holds
+        /// </summary>
+        /// [Obsolete("This property is deprecated, please use byte[] PasswordData instead.")]
+        [NotNull]
         public string Password { get; }
 
         /// <summary>
         /// Gets the password that this object holds
         /// </summary>
         [NotNull]
-        public string Username { get; }
+        public byte[] PasswordData { get; }
```

Comment:
This is usually the job of the `SecureString` class

Reply:
SecureSring works with android and ios too?

Reply:
I assume so.  I don't see any reason why it wouldn't.  Do you have information suggesting that it doesn't?

Reply:
Checked, SecureString Class works on all platforms.